### PR TITLE
Refactoring DataBinder YAML interfaces

### PR DESCRIPTION
**Pull request by ponyisi**
Strawman proposal for refactoring DataBinder YAML interfaces for compatibility with Python API

### CODE REVIEW

File: examples/bigger_uproot.py
Comment (ponyisi) [inline examples/bigger_uproot.py:1]: Demonstrate ServiceX configuration with pure dictionary (reduces imports, easier to manipulate from code)

File: examples/config_databinder_func_adl.yaml
Comment (ponyisi) [inline examples/config_databinder_func_adl.yaml:1]: Move to &/* syntax

File: examples/config_databinder_python.yaml
Comment (ponyisi) [inline examples/config_databinder_python.yaml:1]: Move to &/* syntax (definitions must come first)

File: examples/single_file_uproot.py
Comment (ponyisi) [inline examples/single_file_uproot.py:1]: Make backend choice uniform among examples

File: tests/test_databinder.py
Comment (ponyisi) [inline tests/test_databinder.py:1]: Add tests for YAML features

**Review by kyungeonchoi — COMMENTED** [review]
Looks good to me! Thank you!

**Pull request merged**